drivers/shared/validators/validators.go

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package validators
+
+import (
+	"errors"
+	"fmt"
+	"os/user"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/nomad/client/lib/idset"
+	"github.com/hashicorp/nomad/client/lib/numalib/hw"
+)
+
+var (
+	ErrInvalidBound = errors.New("range bound not valid")
+	ErrEmptyRange   = errors.New("range value cannot be empty")
+	ErrInvalidRange = errors.New("lower bound cannot be greater than upper bound")
+)
+
+type validator struct {
```

Comment:
Do we want to export this type considering it is an used by other packages with exported functions?

Reply:
what do you mean?